File: python/armstrong-numbers/armstrong_numbers.py
def is_armstrong_number(number):
    if number == 0:
        return True

    arr = []
    number_cpy = number
    while number_cpy / 10 > 0:
        arr.append(number_cpy % 10)
        number_cpy = number_cpy // 10

    print(arr)
    sum = 0
    for n in arr:
        sum += pow(n, len(arr))

    print(sum)
    return sum == number

File: zig/armstrong-numbers/armstrong_numbers.zig
const std = @import("std");

pub fn isArmstrongNumber(num: u128) bool {
    var n = num;
    var l: usize = 0;
    while (n > 0) {
        n /= 10;
        l += 1;
    }

    var sum: u128 = 0;
    n = num;
    while (n > 0) { 
        const k = n%10;
        var product: u128 = 1;
        for (0..l) |_| {
            product *= k;
        }
        sum += product;
        n /= 10;
    }

    return sum == num;
}
